src/activity-logger/background/log-page-visit.js

```diff
@@ -23,26 +24,32 @@ async function storeVisit({ timestamp, url, page }) {
  *  to the new page.
  * @param {any} visit The new visit to add to index; should occur regardless of reidentify outcome.
  */
-async function updateIndex(finalPagePromise, visit) {
-    // Wait until all page analyis is done
-    const { page } = await finalPagePromise
+async function updateIndex(storePageResult, visit, pageId) {
+    if (storePageResult == null) {
+        index.addMetaConcurrent('visit', Date.now(), pageId)
```

Comment:
Should have an `await` or `return` on this call, else the `Promise` returned from this function (`updateIndex`) will resolve before finished. Probably better to `await` to be consistent with the other indexing call in L41